controllers/renovate_util.go

```diff
@@ -268,27 +308,127 @@ func (u ComponentDependenciesUpdater) GetUpdateTargetsGithubApp(ctx context.Cont
-			Endpoint:                git.BuildAPIEndpoint("github").APIEndpoint("github.com"),
-			Repositories:            repositories,
-			ImageRepositoryHost:     imageRepositoryHost,
-			ImageRepositoryUsername: imageRepositoryUsername,
-			ImageRepositoryPassword: imageRepositoryPassword,
+			ComponentName:                  component.Name,
+			ComponentCustomRenovateOptions: customRenovateOptions,
+			GitProvider:                    gitProvider,
+			Username:                       fmt.Sprintf("%s[bot]", slug),
+			GitAuthor:                      fmt.Sprintf("%s <126015336+%s[bot]@users.noreply.github.com>", slug, slug),
+			Token:                          githubAppInstallation.Token,
+			Endpoint:                       git.BuildAPIEndpoint("github").APIEndpoint("github.com"),
+			Repositories:                   repositories,
+			ImageRepositoryHost:            imageRepositoryHost,
+			ImageRepositoryUsername:        imageRepositoryUsername,
+			ImageRepositoryPassword:        imageRepositoryPassword,
 		})
 		log.Info("component to update for installations", "component", component.Name, "repositories", repositories)
 	}
 
 	return targetsToUpdate
 }
 
+// ReadCustomRenovateConfigMaps This method returns custom renovate options from config map
+func (u ComponentDependenciesUpdater) ReadCustomRenovateConfigMap(ctx context.Context, component *v1alpha1.Component) (CustomRenovateOptions, error) {
+	log := logger.FromContext(ctx)
+	customRenovateOptions := CustomRenovateOptions{}
+	customComponentRenovateConfigMap := readComponentCustomRenovateConfigMapAnnotation(component)
+
+	customRenovateConfigMapName := NamespaceWideRenovateConfigName
+	if customComponentRenovateConfigMap != "" {
+		customRenovateConfigMapName = customComponentRenovateConfigMap
+	}
+
+	customRenovateConfigMap := &corev1.ConfigMap{}
+	log.Info("trying to get custom renovate config map", "configMapName", customRenovateConfigMapName)
+	if err := u.Client.Get(ctx, types.NamespacedName{Name: customRenovateConfigMapName, Namespace: component.Namespace}, customRenovateConfigMap); err != nil {
+		if errors.IsNotFound(err) {
+			if customRenovateConfigMapName != NamespaceWideRenovateConfigName {
+				log.Info("custom renovate config map in component annotation doesn't exist", "configMapName", customRenovateConfigMapName)
+			}
+			// no need to fail if custom renovate config map doesn't exist
+			return customRenovateOptions, nil
+		}
+		return customRenovateOptions, err
+	}
+	log.Info("will use custom renovate config map", "configMapName", customRenovateConfigMapName)
+
+	automergeOption, automergeExists := customRenovateConfigMap.Data[RenovateConfigMapAutomergeKey]
+	if automergeExists {
+		automergeValue, err := strconv.ParseBool(automergeOption)
+		if err != nil {
+			log.Error(err, "can't parse automerge option in configmap", "configMapName", customRenovateConfigMapName, "automergeValue", automergeOption)
+		} else {
+			customRenovateOptions.Automerge = automergeValue
+		}
+	}
```

Comment:
Does it make sense to have a func to parse a boolean value?

Reply:
it wouldn't change anything , I'd have to call that function which would call ParseBool

and then do exactly the same, check err and assign